### ISSUE TITLE
incorrect "Missing type annotation for `U`." flow error triggered by adding `export`.

### ISSUE THREAD

**Issue by DianaSuvorova** [facebook/flow#6828]
Here is a [try flow](https://flow.org/try/#0PQKgBAAgZgNg9gdzCYAodAXAngBwKZgAqAFgJYB2A5mALxgDeqYYOATnDgFxgDOGrFSqgC+AbnTBgYcnDB5W7VjzCooAV3IBjDKTjkwAWywAxDdt3kAFBjJUe3AIIKAhlgA8JQQD4AlAyZgrHgYaqz6NoI8AHQGzjiWzrReYM4+4sISUgB6YACypDw8gmDY+CnkMhjOOnpgUHCsYAAGAKpNUXIKDah4AB44DRh1ZjX6RqZao9a2lPZgTqyuHjO+-sxBIWElM9Gx8Yk0yanpQA)

```js
/* @flow */

type Thing = {
  prop: string
};

// no errors 
function myFunction(things: Array<Thing>) {
  return things.map(a => a);
}

// Missing type annotation for `U`. error
export function myFunction(things: Array<Thing>) {
  return things.map(a => a);
}
```

**Comment by swac** [comment]
It might be referring to the generic `U` in `map`'s definition: https://github.com/facebook/flow/blob/master/lib/core.js#L247.

Either changing the call to `things.map<Thing>(a => a)` or specifying `myFunction`'s return type as `Array<Thing>` should fix it.

**Comment by swac** [comment]
I think it's referring to the generic `U` in `map`'s definition: https://github.com/facebook/flow/blob/master/lib/core.js#L247.

Changing the call to `things.map<Thing>(a =>a)` or specifying a return type of `Array<Thing>` for `myFunction` seems to fix it. It does seem weird that it only shows the error when the `export` is present though.

**Comment by mdugue** [comment]
got this too recently, not sure what that means

**Comment by jbrown215** [comment]
Flow requires annotations on types in input positions reachable through exports. Until now, we were missing a lot of these cases. This is an example of a case we started catching recently. Even more recent changes, which will be shipped in v0.81.0, will hopefully make these errors clearer. I am also currently working on documentation for fixing these new errors when they appear.

**Comment by dnalborczyk** [comment]
just got this exception. for anyone else, explained here in more detail: https://medium.com/flow-type/asking-for-required-annotations-64d4f9c1edf8
